src/pynxtools/nomad/superclasses_server.py

```diff
@@ -0,0 +1,53 @@
+
+######################################################################
+######################## import libraries ############################
+######################################################################
+from fastapi import FastAPI, HTTPException
+from owlready2 import get_ontology, sync_reasoner
+from owlready2.namespace import Ontology
+import os
+from fastapi.responses import RedirectResponse
+
+#######################################################################
+########################## define app and functions ###################
+#######################################################################
+app = FastAPI()
+
+def load_ontology(*args) -> Ontology:
+    owl_file = os.path.join(os.path.dirname(__file__), *args)
+    return get_ontology(owl_file).load()
+
+def fetch_superclasses(ontology, class_name):
+    # Search for the first IRI ending with the class name
+    cls = ontology.search_one(iri="*" + class_name)
+    if cls is None:
+        raise ValueError(f"Class '{class_name}' not found in the ontology.")
+    all_superclasses =  cls.ancestors()
+    #filter unwantedsuperclasses
+    unwanted_superclasses = {
```

Comment:
please collect this set from the ontology itself by asking for the superclasses of a specific (list  of) unwanted element(s) (e.g. all superclasses of "NeXusOntology_full.NeXusApplicationClass").